ia/src/ml_agent.py

```
@@ -122,63 +130,128 @@ def analyze_vision(client):
+  inventory = client.get("inventory", {})
+  missing_resources = {}
+
+  valid_resources = ["linemate", "deraumere", "sibur", "mendiane", "phiras", "thystame"]
+
+  for resource, quantity in needed.items():
+    if inventory.get(resource, 0) < quantity and resource in valid_resources:
+      missing_resources[resource] = quantity - inventory.get(resource, 0)
+
+  return missing_resources
+
+def verify_incantation(client, current_cell, needed_resources):
```

Ruff (0.11.9)

Missing return type annotation for public function `verify_incantation`

Add return type annotation: `None`

(ANN201)